### ISSUE TITLE
Option to format indices in generated properties

### ISSUE THREAD

**Issue by jakub-bochenski** [mikefarah/yq#1933]
**Please describe your feature request.**
I want the generated properties to work with Spring Boot.

Problem is Spring Boot uses a bracket notation for list indices (`[]`).

**Describe the solution you'd like**
If we have data:

```yaml
# block comments come through
person: # neither do comments on maps
    name: Mike Wazowski # comments on values appear
    pets: 
    - cat # comments on array values appear
    food: [pizza] # comments on arrays do not
emptyArray: []
emptyMap: []
```

it could output

```yaml
# block comments come through
# comments on values appear
person.name = Mike Wazowski
​
# comments on array values appear
person.pets[0]=cat
person.food[1]=pizza
```

**Describe alternatives you've considered**
Sed the output

**Comment by jakub-bochenski** [comment]
FTR sed solution to compact spaces and bracked indices is:

```
find \
  -name 'application*.yml' \
  -exec sh -c "yq -o=props {} | sed -r  -e 's/^([^=]+)\.([0-9]+)/\1[\2]/g' -e 's/^([^=]+) = /\1=/g' > \$(echo {} | sed s/.yml$/.properties/)" \; \
  -delete
```

**Comment by mikefarah** [comment]
Fixed in 4.42.1
